### PR TITLE
Clears up gun and mob related hard dels

### PR DESCRIPTION
**Pull request by carlarctg**
# About the pull request

Clears up gun and mob related hard dels

guaranteed not to randomly delete shit this time

# Explain why it's good for the game

hard dels LAG the game !

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
code: Clears up gun and mob related hard dels
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 /mob/living/carbon/Destroy()
 	stomach_contents?.Cut()
-
+	QDEL_NULL(view_change_sources)
```
Comment (harryob) [inline code/modules/mob/living/carbon/carbon.dm:19]: just null the reference to the list. if no one's holding a reference to it, it'll be GCd anyway

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/carbon.dm (hunk 1 above)
Comment (carlarctg) [reply code/modules/mob/living/carbon/carbon.dm:19]:
```suggestion
	view_change_sources = null
```

**Comment by carlarctg** [comment]
bro draft prs!

**Comment by harryob** [comment]
> bro draft prs!

look at your github notifications

File: code/modules/projectiles/guns/energy.dm [hunk 2]
```diff
@@ -33,6 +33,12 @@
 	update_icon()
 	max_shots = round((cell.maxcharge / charge_cost), 1)
 
+/obj/item/weapon/gun/energy/Destroy()
+	QDEL_NULL(cell)
+	. = ..()
+
+
+
```
Comment (harryob) [inline code/modules/projectiles/guns/energy.dm:41]:
```suggestion
```

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/gun.dm [hunk 3]
```diff
@@ -266,6 +266,27 @@
...
+	ammo = null
+	current_mag = null
+	target = null
+	last_moved_mob = null
+	if(flags_gun_features & GUN_FLASHLIGHT_ON)//Handle flashlight.
+		flags_gun_features &= ~GUN_FLASHLIGHT_ON
+		if(ismob(loc))
+			for(var/slot in attachments)
+				var/obj/item/attachable/R = attachments[slot]
+				if(!R) continue
```
Comment (harryob) [inline code/modules/projectiles/gun.dm:280]:
```suggestion
				var/obj/item/attachable/potential_attachment = attachments[slot]
				if(!potential_attachment)
					continue
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by carlarctg** [comment]
stop farming reviews dude

**Review by harryob — APPROVED** [review]

**Pull request merged**